integration-tests/utils/testutils.go

```diff
@@ -116,9 +116,9 @@ func CheckMetricsEndpointResult(httpAddr string, success bool) error {
 	return nil
 }
 
-// ConnectToCluster is used to connect to source and destination clusters
-func ConnectToCluster(hostname string, username string, password string, port int) (*gocql.Session, error) {
+func ConnectToClusterUsingVersion(hostname string, username string, password string, port int, protoVersion int) (*gocql.Session, error) {
 	cluster := NewCluster(hostname, username, password, port)
+	cluster.ProtoVersion = protoVersion
```

Comment:
keep in mind this skips protocol version negotiation on gocql.

Reply:
Hmm, without this connection to Simulacron fails in `Cluster2.1_MaxCCProtoVer4_NegotiatedProtoVer3`.

Reply:
I did not test it end-to-end, but looking at GoCQL code, it seems it does not support protocol version negotiation. See https://github.com/gocql/gocql/blob/master/control.go#L219. It tries always V4. Error raised with Simulacron test is: `gocql: unable to create session: unable to discover protocol version: unexpected protocol version in response: got 3 expected 4`. Simulacron returns V3, but driver does not automatically downgrade.

Reply:
gocql absolutely supports protocol version negotiation but you're skipping it by telling gocql to use a specific version, you should remove that from that test